In [1]:
from google.colab import drive
drive.mount('/content/drive')
import os
os.chdir('/content/drive/My Drive/AI/BERT-Tweet-Sentiment-Extraction')

Mounted at /content/drive


# Cài đặt các thư viện cần thiết

In [2]:
!pip3 install tokenizers==0.8.1rc2

     |████████████████████████████████| 3.0MB 4.3MB/s 


In [3]:
!pip3 install transformers==3.3.1

     |████████████████████████████████| 1.1MB 4.6MB/s 
     |████████████████████████████████| 1.1MB 52.9MB/s 
     |████████████████████████████████| 890kB 45.3MB/s 
  Created wheel for sacremoses: filename=sacremoses-0.0.43-cp36-none-any.whl size=893257 sha256=1dbfa8158a6ff8374c353381233439872d8217ed25e313f8222eae54075fb94a
  Stored in directory: /root/.cache/pip/wheels/29/3c/fd/7ce5c3f0666dab31a50123635e6fb5e19ceb42ce38d4e58f45
Successfully built sacremoses


# Chạy với Bert

In [ ]:
!python3 ./BertRun.py

2020-10-26 15:57:26.627556: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
========== Fold 1 ========== 
100% 21984/21984 [30:50<00:00, 11.88it/s]
Epoch 1/3 | train | Loss: 2.3266 | Jaccard: 0.6499
100% 5497/5497 [02:29<00:00, 36.82it/s]
Epoch 1/3 |  val  | Loss: 1.8151 | Jaccard: 0.6903
100% 21984/21984 [30:47<00:00, 11.90it/s]
Epoch 2/3 | train | Loss: 1.7642 | Jaccard: 0.7078
100% 5497/5497 [02:29<00:00, 36.69it/s]
Epoch 2/3 |  val  | Loss: 1.8477 | Jaccard: 0.6970
100% 21984/21984 [31:53<00:00, 11.49it/s]
Epoch 3/3 | train | Loss: 1.4093 | Jaccard: 0.7525
100% 5497/5497 [02:42<00:00, 33.92it/s]
Epoch 3/3 |  val  | Loss: 2.1386 | Jaccard: 0.6949
========== Fold 2 ========== 
100% 21985/21985 [33:08<00:00, 11.06it/s]
Epoch 1/3 | train | Loss: 2.2869 | Jaccard: 0.6492
100% 5496/5496 [02:43<00:00, 33.65it/s]
Epoch 1/3 |  val  | Loss: 1.8421 | Jaccard: 0.6889
100% 21985/21985 [33:15<00:00, 11.02it/s]
Epoch 2/3 | trai

# Chạy với Roberta

In [ ]:
!python3 ./RobertaRun.py

2020-10-27 07:40:29.300453: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
========== Fold 1 ========== 
100% 21984/21984 [32:00<00:00, 11.45it/s]
Epoch 1/3 | train | Loss: 2.6045 | Jaccard: 0.6426
100% 5497/5497 [01:38<00:00, 55.64it/s]
Epoch 1/3 |  val  | Loss: 1.7709 | Jaccard: 0.6998
100% 21984/21984 [31:30<00:00, 11.63it/s]
Epoch 2/3 | train | Loss: 1.8547 | Jaccard: 0.6932
100% 5497/5497 [01:40<00:00, 54.75it/s]
Epoch 2/3 |  val  | Loss: 1.8011 | Jaccard: 0.7073
100% 21984/21984 [31:26<00:00, 11.65it/s]
Epoch 3/3 | train | Loss: 1.6343 | Jaccard: 0.7200
100% 5497/5497 [01:39<00:00, 55.08it/s]
Epoch 3/3 |  val  | Loss: 1.8975 | Jaccard: 0.7068
========== Fold 2 ========== 
100% 21985/21985 [31:25<00:00, 11.66it/s]
Epoch 1/3 | train | Loss: 2.6602 | Jaccard: 0.6376
100% 5496/5496 [01:39<00:00, 55.15it/s]
Epoch 1/3 |  val  | Loss: 1.8018 | Jaccard: 0.7049
100% 21985/21985 [31:27<00:00, 11.65it/s]
Epoch 2/3 | trai

========== Fold 4 ========== 
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 21985/21985 [1:19:32<00:00,  4.61it/s]
Epoch 1/3 | train | Loss: 2.6258 | Jaccard: 0.6414
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5496/5496 [03:15<00:00, 28.07it/s]
Epoch 1/3 |  val  | Loss: 1.8213 | Jaccard: 0.6876
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 21985/21985 [1:19:52<00:00,  4.59it/s]
Epoch 2/3 | train | Loss: 1.8606 | Jaccard: 0.6932
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5496/5496 [03:14<00:00, 28.32it/s]
Epoch 2/3 |  val  | Loss: 1.8303 | Jaccard: 0.7048
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 21985/21985 [1:21:22<00:00,  4.50it/s]
Epoch 3/3 | train | Loss: 1.6162 | Jaccard: 0.7195
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5496/5496 [03:17<00:00, 27.78it/s]
Epoch 3/3 |  val  | Loss: 1.9853 | Jaccard: 0.6922
========== Fold 5 ========== 
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 21985/21985 [1:20:57<00:00,  4.53it/s]
Epoch 1/3 | train | Loss: 2.5423 | Jaccard: 0.6412
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5496/5496 [03:17<00:00, 27.86it/s]
Epoch 1/3 |  val  | Loss: 1.7811 | Jaccard: 0.7026
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 21985/21985 [1:21:12<00:00,  4.51it/s]
Epoch 2/3 | train | Loss: 1.8484 | Jaccard: 0.6907
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5496/5496 [03:29<00:00, 26.26it/s]
Epoch 2/3 |  val  | Loss: 1.7527 | Jaccard: 0.7048
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 21985/21985 [1:25:32<00:00,  4.28it/s]
Epoch 3/3 | train | Loss: 1.5910 | Jaccard: 0.7230
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5496/5496 [03:30<00:00, 26.13it/s]
Epoch 3/3 |  val  | Loss: 1.8409 | Jaccard: 0.7021

# Chạy với Albert

Albert Base V1

In [ ]:
!python3 ./AlbertRun.py

2020-10-27 20:22:55.424760: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
========== Fold 1 ========== 
100% 21984/21984 [15:47<00:00, 23.21it/s]
Epoch 1/3 | train | Loss: 2.4660 | Jaccard: 0.6407
100% 5497/5497 [01:36<00:00, 57.05it/s]
Epoch 1/3 |  val  | Loss: 1.9092 | Jaccard: 0.6871
100% 21984/21984 [16:03<00:00, 22.82it/s]
Epoch 2/3 | train | Loss: 1.9635 | Jaccard: 0.6814
100% 5497/5497 [01:38<00:00, 55.69it/s]
Epoch 2/3 |  val  | Loss: 1.8603 | Jaccard: 0.6914
100% 21984/21984 [15:46<00:00, 23.24it/s]
Epoch 3/3 | train | Loss: 1.7881 | Jaccard: 0.6974
100% 5497/5497 [01:39<00:00, 55.09it/s]
Epoch 3/3 |  val  | Loss: 1.8563 | Jaccard: 0.6882
========== Fold 2 ========== 
100% 21985/21985 [16:05<00:00, 22.77it/s]
Epoch 1/3 | train | Loss: 2.5533 | Jaccard: 0.6401
100% 5496/5496 [01:43<00:00, 52.97it/s]
Epoch 1/3 |  val  | Loss: 1.8565 | Jaccard: 0.6873
100% 21985/21985 [16:23<00:00, 22.36it/s]
Epoch 2/3 | trai

Albert Base V2

In [5]:
!python3 ./AlbertRun.py

2020-10-27 23:39:15.916308: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
========== Fold 1 ========== 
100% 21984/21984 [15:58<00:00, 22.94it/s]
Epoch 1/3 | train | Loss: 2.8810 | Jaccard: 0.6110
100% 5497/5497 [01:39<00:00, 55.39it/s]
Epoch 1/3 |  val  | Loss: 1.8990 | Jaccard: 0.6882
100% 21984/21984 [16:00<00:00, 22.89it/s]
Epoch 2/3 | train | Loss: 2.0329 | Jaccard: 0.6749
100% 5497/5497 [01:38<00:00, 55.84it/s]
Epoch 2/3 |  val  | Loss: 1.8436 | Jaccard: 0.6965
100% 21984/21984 [15:58<00:00, 22.94it/s]
Epoch 3/3 | train | Loss: 1.8557 | Jaccard: 0.6917
100% 5497/5497 [01:38<00:00, 56.01it/s]
Epoch 3/3 |  val  | Loss: 1.8334 | Jaccard: 0.7028
========== Fold 2 ========== 
100% 21985/21985 [16:00<00:00, 22.88it/s]
Epoch 1/3 | train | Loss: 3.0015 | Jaccard: 0.6118
100% 5496/5496 [01:40<00:00, 54.61it/s]
Epoch 1/3 |  val  | Loss: 1.9716 | Jaccard: 0.6786
100% 21985/21985 [15:59<00:00, 22.91it/s]
Epoch 2/3 | trai